app/src/main/java/org/mifos/mobile/ui/beneficiary_detail/BeneficiaryDetailScreen.kt

```diff
@@ -0,0 +1,252 @@
+        }
+    )
+}
+
+@Composable
+fun BeneficiaryDetailScreen(
+    beneficiary: Beneficiary?,
+    uiState: BeneficiaryUiState,
+    navigateBack: () -> Unit,
+    updateBeneficiary: () -> Unit,
+    deleteBeneficiary: (id: Long?) -> Unit,
+) {
+    var openDropdown by rememberSaveable { mutableStateOf(false) }
+    var showAlert by rememberSaveable { mutableStateOf(false) }
+    val context = LocalContext.current
+    var deleteButtonClicked by remember { mutableStateOf(false) }
+    var showBeneficiaryNullError by rememberSaveable {
+        mutableStateOf(true)
+    }
+
+    if (showAlert) {
+        MifosAlertDialog(
+            onDismissRequest = {
+                showAlert = !showAlert
+            },
+            dismissText = context.getString(R.string.cancel),
+            onConfirmation = {
+                showAlert = !showAlert
+                deleteButtonClicked = true
+                deleteBeneficiary.invoke(beneficiary?.id?.toLong())
+            },
+            confirmationText = context.getString(R.string.delete),
+            dialogTitle = context.getString(R.string.delete_beneficiary),
+            dialogText = context.getString(R.string.delete_beneficiary_confirmation) + "?"
+        )
+    }
+
+    Scaffold(
+        topBar = {
+            BeneficiaryDetailTopAppBar(
+                navigateBack = navigateBack,
+                updateBeneficiaryClicked = updateBeneficiary,
+                updateDropdownValue = { value ->
+                    openDropdown = value
+                    openDropdown
+                },
+                showAlert = {
+                    showAlert = true
+                }
+            )
+        }
+    ) {
+        Box(
+            modifier = Modifier.padding(it)
+        ) {
+            BeneficiaryDetailContent(
+                beneficiary = beneficiary
+            )
+
+            when (uiState) {
+                BeneficiaryUiState.Initial -> Unit
+
+                BeneficiaryUiState.Loading -> {
+                    MifosProgressIndicatorOverlay()
+                }
+
+                BeneficiaryUiState.DeletedSuccessfully -> {
+                    Toast.makeText(
+                        context,
+                        stringResource(id = R.string.beneficiary_deleted_successfully),
+                        Toast.LENGTH_SHORT
+                    ).show()
+                    navigateBack.invoke()
+                }
+
+                is BeneficiaryUiState.ShowError -> {
+                    if (deleteButtonClicked) {
+                        Toast.makeText(context, uiState.message, Toast.LENGTH_SHORT).show()
+                        deleteButtonClicked = false
+                    }
+                    if (showBeneficiaryNullError) {
+                        Toast.makeText(context, uiState.message, Toast.LENGTH_SHORT).show()
+                        showBeneficiaryNullError = false
+                    }
+                }
+
+                else -> Unit
```

Comment:
we dont have to use else. Make a new UIState class specific to only this screen with above 4 subclasses. Then we can eliminate else case.